sklearn/preprocessing/_target_encoder.py

```diff
@@ -0,0 +1,249 @@
+        used to encoding the other split. Afterwards, a final categorical encoding is
+        obtained from all the training data, which is used to encode data
+        during :meth:`transform`. This means that `fit().transform()` does not equal
+        `fit_transform()`.
+
+    .. versionadded:: 1.3
+
+    Parameters
+    ----------
+    categories : 'auto' or a list of array-like, default='auto'
+        Categories (unique values) per feature:
+
+        - 'auto' : Determine categories automatically from the training data.
+        - list : ``categories[i]`` holds the categories expected in the ith column. The
+          passed categories should not mix strings and numeric values within a single
+          feature, and should be sorted in case of numeric values.
+
+        The used categories can be found in the ``categories_`` attribute.
+
+    target_type : {"auto", "continuous", "binary"}, default="auto"
+        Type of target
+
+        - `"auto"` : Type of target is inferred with
+          :func:`~sklearn.utils.multiclass.type_of_target`
+        - `"continuous"` : Continious target
+        - `"binary"` : Binary target
+
+    smooth : float, default=30.0
+        The amount of mixing the categorical encoding with the global target mean. A
+        larger `smooth` value will put more weight on the global target mean.
+
+    cv : int, cross-validation generator or an iterable, default=None
+        Determines the cross-validation splitting strategy used in
+        :meth:`fit_transform`. Possible inputs for cv are:
+
+        - None, to use the default 5-fold cross validation,
+        - integer, to specify the number of folds in a `(Stratified)KFold`,
+        - :term:`CV splitter`,
+        - An iterable yielding (train, test) splits as arrays of indices.
+
+        Refer :ref:`User Guide <cross_validation>` for the various cross-validation
+        strategies that can be used here.
+
+    Attributes
+    ----------
+    encodings_ : list of shape (n_features,) of ndarray
+        For feature `i`, `encodings_[i]` is the encoding matching the
+        categories listed in `categories_[i]`.
+
+    categories_ : list of shape (n_features,) of ndarray
+        The categories of each feature determined during fitting
+        (in order of the features in X and corresponding with the output
+        of :meth:`transform`).
+
+    target_type_ : str
+        Type of target.
+
+    encoding_mean_ : float
+        The overall mean of the target.
+
+    n_features_in_ : int
+        Number of features seen during :term:`fit`.
+
+    feature_names_in_ : ndarray of shape (`n_features_in_`,)
+        Names of features seen during :term:`fit`. Defined only when `X`
+        has feature names that are all strings.
+
+    See Also
+    --------
+    OrdinalEncoder : Performs an ordinal (integer) encoding of the categorical features.
+    OneHotEncoder : Performs a one-hot encoding of categorical features.
```

Comment:
We should cross-reference `TargetEncoder` from onehot's and ordinal's docstring as well